### ISSUE TITLE
Validate forward pass output against llama.cpp reference (1e-4 tolerance)

### ISSUE THREAD

**Issue by SmartEst74** [SmartEst74/PowerInfer_x64#127]
Priority: HIGH. Run the same input through our forward pass and llama.cpp, compare logits. Must match within 1e-4.
